src/lib/game/world/map.js

```diff
@@ -130,12 +130,17 @@ class WorldMap extends THREE.Group {
         entry.position.y
       );
 
+      // Provided as (Z, X, -Y)
       m2.rotation.set(
+        entry.rotation.z * Math.PI / 180,
         entry.rotation.x * Math.PI / 180,
-        -entry.rotation.z * Math.PI / 180,
-        entry.rotation.y * Math.PI / 180
+        -entry.rotation.y * Math.PI / 180
       );
 
+      // Adjust M2 rotation to match Wowser's axes.
+      const quat = m2.quaternion;
+      quat.set(quat.x, quat.y, quat.z, -quat.w);
+
```

Comment:
This is the most straight-forward way of adjusting the rotation to fit Wowser :+1: